### ISSUE TITLE
Level 7_2

### ISSUE THREAD

**Issue by tzk00** [nus-cs2030/2122-s2#535]
## Screenshots (if any): 

<img width="523" alt="Screenshot 2022-04-24 at 12 58 47 PM" src="https://user-images.githubusercontent.com/99874913/164957356-cd99974a-1628-420d-99e1-55a04370bf6c.png">
<img width="289" alt="Screenshot 2022-04-24 at 1 02 01 PM" src="https://user-images.githubusercontent.com/99874913/164957413-bdc23277-3081-4319-b6f8-ccd7bf76e6f3.png">
The times above are the rest times for the respective serve events.

##Description:
Hi everyone, I am having trouble passing test case 7_2. Customer 10 is being served at 10.781 instead of 11.636. I have already checked if eventTime >= serverNextAvailTime. I think there might be a problem with my rest times but I can't seem to figure out what is wrong.

**Comment by pearlynliu** [comment]
hi! i think you might want to refer to #532!
hope this helps

**Comment by jonyeokj** [comment]
I believe this is because you are calling the get() from the Supplier<Double> variable restTimes wrongly, so it's not giving you the rest time that the server should have before serving 10.

**Comment by jamietan123** [comment]
Rest time should only be called when Done

**Comment by kh1js** [comment]
Yup I think you can get the restTime from the supplier only once. (ie only call it when necessary)

Otherwise you can check if the server is not resting before serving the customer.

Hope it helps!

**Comment by nicljr** [comment]
Hi, maybe you can check your Done Event class that it's getting the rest time only once after the Server is finished serving the customer. I made a new Event for the Server to finish resting and make it free to serve the next customer and it made my implementation much easier.

**Comment by ReganKvioke** [comment]
I agree with @nicljr by only getting rest time after the Server is finished serving the customer, then we can move on to another Event that will be given out when Server is finish resting. Just take note to update the rest time to make sure that your statistics updates

**Comment by zendenleong** [comment]
Hi, your restTime should only be called after every Done event. For myself, I only called get() after every done event in my simulate to get the resttime for the server.

**Comment by tristky** [comment]
Hi, im also stuck on this level, and although after reading the advice, i am still not able to figure this out, but i will continue trying, thanks
